kolibri/plugins/user/assets/src/views/sign-in-page/facility-modal.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+
+  <core-modal @cancel="emitClose" :title="$tr('facilitySelectionModalHeader')">
+    <form @submit.prevent="submitAndClose">
+      {{ $tr('facilitySelectionPrompt') }}
+
+      <k-radio-button
+        v-for="facility in facilities"
+        v-model="selectedFacility"
+        :key="facility.id"
+        :label="facility.name"
+        :radiovalue="facility.id"
+      />
+      <div class="core-modal-buttons">
+        <k-button
+          :text="$tr('submitFacilitySelectionButtonPrompt')"
+          :primary="true"
+          type="submit"
+        />
+      </div>
+    </form>
+  </core-modal>
+
+</template>
+
+
+<script>
+
+  import coreModal from 'kolibri.coreVue.components.coreModal';
+  import kRadioButton from 'kolibri.coreVue.components.kRadioButton';
+  import kButton from 'kolibri.coreVue.components.kButton';
+  import { getFacilityConfig } from 'kolibri.coreVue.vuex.actions';
+
+  export default {
+    name: 'facilityModal',
+    components: {
+      coreModal,
+      kRadioButton,
+      kButton,
+    },
+    data() {
+      return {
+        selectedFacility: this.currentFacilityId,
+      };
+    },
+    methods: {
+      emitClose() {
+        this.$emit('close');
+      },
+      submitAndClose() {
+        this.setFacilityId();
+        this.clearLoginError();
+        this.getFacilityConfig(this.selectedFacility);
```

Comment:
Also, we shouldn't clear the modal until the new facility config has been loaded and set, otherwise it may produce a confusing user experience? If the facility config for the newly selected facility is different, this could cause the interface to change mid-user input if the response is slow.